src/oid.rs

```diff
@@ -153,10 +153,16 @@ impl ObjectId {
 
     // Generate a random 5-byte array.
     fn gen_process_id() -> [u8; 5] {
-        let rng = thread_rng().gen_range(0, MAX_U24) as u32;
-        let mut buf: [u8; 5] = [0; 5];
-        buf[0..4].copy_from_slice(&rng.to_be_bytes());
-        buf
+        lazy_static! {
```

Comment:
When does the code in the `lazy_static` block actually execute?

Reply:
It will execute the first time that the reference it defines (i.e. `BUF`) is dereferenced, which will be the first this the method is called. [`lazy_static`'s documentation](https://docs.rs/lazy_static/1.4.0/lazy_static/#semantics) has more details on this.